modules/api/test/LorisApiImagesTest.php

```diff
@@ -0,0 +1,922 @@
+     *
+     * @return void
+     */
+    public function tearDown()
+    {
+        parent::tearDown();
+    }
+    /**
+     * Tests the HTTP GET request for the
+     * endpoint /candidates/{candid}/{visit}/images
+     *
+     * @return void
+     */
+    public function testGetCandidatesCandidVisitImages(): void
+    {
+        $response = $this->client->request(
+            'GET',
+            "candidates/$this->candidTest/$this->visitTest/images/",
+            [
+                'headers'     => $this->headers,
+                'http_errors' => false
+            ]
+        );
+        $this->assertEquals(200, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        $imagesArray = json_decode(
+            (string) utf8_encode(
+                $response->getBody()->getContents()
+            ),
+            true
+        );
+
+        $this->assertSame(gettype($imagesArray), 'array');
+        $this->assertSame(gettype($imagesArray['Meta']), 'array');
+        $this->assertSame(gettype($imagesArray['Meta']['CandID']), 'string');
+        $this->assertSame(gettype($imagesArray['Meta']['Visit']), 'string');
+        $this->assertSame(gettype($imagesArray['Files']), 'array');
+        $this->assertSame(
+            gettype($imagesArray['Files']['0']['OutputType']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($imagesArray['Files']['0']['Filename']),
+            'string'
+        );
+        $this->assertSame(
+            gettype($imagesArray['Files']['0']['AcquisitionType']),
+            'string'
+        );
+
+        $this->assertArrayHasKey('Meta', $imagesArray);
+        $this->assertArrayHasKey('CandID', $imagesArray['Meta']);
+        $this->assertArrayHasKey('Visit', $imagesArray['Meta']);
+        $this->assertArrayHasKey('Files', $imagesArray);
+        $this->assertArrayHasKey('OutputType', $imagesArray['Files']['0']);
+        $this->assertArrayHasKey('Filename', $imagesArray['Files']['0']);
+        $this->assertArrayHasKey('AcquisitionType', $imagesArray['Files']['0']);
+    }
+
+    /**
+     * Tests the HTTP GET request for the
+     * endpoint /candidates/{candid}/{visit}/images/filename
+     *
+     * @return void
+     */
+    public function testGetCandidatesCandidVisitImagesFilename(): void
+    {
+        $resource = fopen($this->imagefileTest, 'w');
+        $stream   = GuzzleHttp\Psr7\stream_for($resource);
+        $response = $this->client->request(
+            'GET',
+            "candidates/$this->candidTest/$this->visitTest/images/" .
+            "$this->imagefileTest",
+            [
+                'headers'     => $this->headers,
+                'http_errors' => false,
+                'save_to'     => $stream
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
```

Comment:
It should exists

Reply:
It exists when the file actually exists. We need to skip to avoid an error, because it is not in the Docker environment. Thus, not skipping causes Travis to fail